src/huggingface_hub/hf_api.py

```diff
@@ -417,6 +418,22 @@ def set_access_token(access_token: str):
     def unset_access_token():
         erase_from_credential_store(USERNAME_PLACEHOLDER)
 
+    def get_model_tags(self) -> ModelTags:
+        "Gets all valid model tags as a nested namespace object"
+        path = f"{api.endpoint}/api/models-tags-by-type"
+        r = requests.get(path)
+        r.raise_for_status()
+        d = r.json()
+        return ModelTags(d)
+
+    def get_dataset_tags(self) -> DatasetTags:
+        "Gets all valid dataset tags as a nested namespace object"
+        path = f"{api.endpoint}/api/datasets-tags-by-type"
+        r = requests.get(path)
+        r.raise_for_status()
+        d = r.json()
+        return DatasetTags(d)
+
```

Comment:
These should also be linked at the bottom of the `HfApi` class:

https://github.com/huggingface/huggingface_hub/blob/73816542dd1a6a7f39d4f4956dd3f7d52b5e79de/src/huggingface_hub/hf_api.py#L1180-L1201

And added to the `__init__` file.